File: doc-examples/hibernate-example-java/src/main/java/example/UserRepository.java

package example;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.annotation.Query;
import io.micronaut.data.annotation.Repository;
import io.micronaut.data.repository.CrudRepository;

import jakarta.validation.constraints.NotNull;
import java.util.List;

@Repository
public interface UserRepository extends CrudRepository<User, Long> {

    @Override
    @Query("UPDATE User SET enabled = false WHERE id = :id")
    void deleteById(@NonNull @NotNull Long id);

    @Query("FROM User user WHERE enabled = false")
    List<User> findDisabled();
}

File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/UserRepository.kt
package example

import io.micronaut.data.annotation.Query
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository
import javax.validation.constraints.NotNull

@JdbcRepository(dialect = Dialect.H2)
interface UserRepository : CrudRepository<User, Long> { // <1>

    @Query("UPDATE user SET enabled = false WHERE id = :id") // <2>
    override fun deleteById(@NotNull id: Long)

    @Query("SELECT * FROM user WHERE enabled = false") // <3>
    fun findDisabled(): List<User>
}
